analyzers/src/SonarAnalyzer.Common/Extensions/AttributeDataExtensions.cs

```diff
@@ -30,5 +32,47 @@ public static class AttributeDataExtensions
 
         public static bool HasAnyName(this AttributeData attribute, params string[] names) =>
             names.Any(x => attribute.HasName(x));
+
+        public static bool TryGetAttributeValue<T>(this AttributeData attribute, string valueName, out T value)
+        {
+            // named arguments take precedence over constructor arguments of the same name. For [Attr(valueName: false, valueName = true)] "true" is returned.
+            if (attribute.NamedArguments.IndexOf(x => x.Key.Equals(valueName, StringComparison.OrdinalIgnoreCase)) is var namedAgumentIndex and >= 0)
+            {
+                return TryConvertConstant(attribute.NamedArguments[namedAgumentIndex].Value, out value);
+            }
+            if (attribute.AttributeConstructor.Parameters.IndexOf(x => x.Name.Equals(valueName, StringComparison.OrdinalIgnoreCase)) is var constructorParameterIndex and >= 0)
+            {
+                return TryConvertConstant(attribute.ConstructorArguments[constructorParameterIndex], out value);
+            }
+            value = default;
+            return false;
+
+            static bool TryConvertConstant(TypedConstant constant, out T value)
```

Comment:
I don't like the `out T value` shadowing the outer one. And setting `out` value from a local method is also not nice. Let's extract this to a private static method.